core_impl/src/main/java/com/google/instrumentation/stats/StatsManagerImpl.java

```diff
@@ -13,21 +13,88 @@
 
 package com.google.instrumentation.stats;
 
+import com.google.common.collect.Multimap;
+import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.stats.View.DistributionView;
+import com.google.instrumentation.stats.View.IntervalView;
+import com.google.instrumentation.stats.ViewDescriptor.DistributionViewDescriptor;
+import com.google.instrumentation.stats.ViewDescriptor.IntervalViewDescriptor;
+import java.util.Collection;
+
 /**
  * Native Implementation of {@link StatsManager}.
  */
 public final class StatsManagerImpl extends StatsManager {
 
+  private final Multimap<MeasurementDescriptor.Name, View> measurementDescriptorToViewMap =
+      MeasurementDescriptorToViewMap.getMap();
   private final StatsContextFactoryImpl statsContextFactory = new StatsContextFactoryImpl();
 
   @Override
   public void registerView(ViewDescriptor viewDescriptor) {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    // We are using a preset measurement {@link RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY}
+    // and view {@link RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW} for this prototype.
+    // The prototype does not allow setting measurement descriptor entries dynamically for now.
+    // TODO(songya): remove the logic for checking preset descriptor.
+    if (!viewDescriptor.equals(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW)) {
+      throw new UnsupportedOperationException(
+          "The prototype will only support Distribution View RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW.");
+    }
+
+    MeasurementDescriptor measurementDescriptor = viewDescriptor.getMeasurementDescriptor();
+    Collection<View> views = measurementDescriptorToViewMap.get(
+        measurementDescriptor.getMeasurementDescriptorName());
+
+    if (views != null) {
+      for (View view : views) {
+        if (view.getViewDescriptor().equals(viewDescriptor)) {
+          throw new IllegalArgumentException("View has already been registered.");
+        }
+      }
+    }
+
+    // TODO(songya): Extend View class and construct View with internal Distributions.
+    View newView;
+    if (viewDescriptor instanceof DistributionViewDescriptor) {
```

Comment:
This set of if/else blocks can use [ViewDescriptor.match(...)](https://github.com/google/instrumentation-java/blob/2e16e06b33d259591117383eb59552ffa15950d2/core/src/main/java/com/google/instrumentation/stats/ViewDescriptor.java#L62) instead of `instanceof` and casting.